src/isle/memoize.py

```diff
@@ -18,7 +18,9 @@ class MemoizeMethod:
     r"""!
     Decorator that memoizes the result of a method call based on given arguments.
 
-    @warning This decorator works only on bound methods not free functions.
+    \warning This decorator works only on bound methods not free functions.
+
+    \note Classes using this decorator on their methods must support hashing.
```

Comment:
Is it possible to detect hash-ability?

Reply:
Kind of: https://stackoverflow.com/questions/3460650/asking-is-hashable-about-a-python-value
The default seems to be fine though: https://stackoverflow.com/questions/11324271/what-is-the-default-hash-in-python

So it seems to be fine for all user defined types (the only place where you can use the decorator).